# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/painting/nsDisplayList.h
# Commit: 1d600b2ed81f
# Full Hash: 1d600b2ed81f163645a033b2e23ee2ac325ef3c1
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-02-02 09:43:12
# Regressor Bug: 1860328
# File Overlap Count: 2
# Description:
#   Bug 1860328 - Fix a bug where caret misses when dragging and dropping r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192377
# ==============================================================================

diff -r 854c462a083b -r 1d600b2ed81f layout/painting/nsDisplayList.h
--- a/layout/painting/nsDisplayList.h	Thu Feb 01 18:01:46 2024 +0000
+++ b/layout/painting/nsDisplayList.h	Thu Feb 01 18:09:39 2024 +0000
@@ -656,7 +656,7 @@
    * Get the frame that the caret is supposed to draw in.
    * If the caret is currently invisible, this will be null.
    */
-  nsIFrame* GetCaretFrame() { return mCaretFrame; }
+  nsIFrame* GetCaretFrame() { return CurrentPresShellState()->mCaretFrame; }
   /**
    * Get the rectangle we're supposed to draw the caret into.
    */
@@ -1726,6 +1726,7 @@
     bool mInsidePointerEventsNoneDoc;
     bool mTouchEventPrefEnabledDoc;
     nsIFrame* mPresShellIgnoreScrollFrame;
+    nsIFrame* mCaretFrame = nullptr;
   };
 
   PresShellState* CurrentPresShellState() {
@@ -1760,7 +1761,6 @@
   // The reference frame for mCurrentFrame.
   const nsIFrame* mCurrentReferenceFrame;
 
-  nsIFrame* mCaretFrame;
   // A temporary list that we append scroll info items to while building
   // display items for the contents of frames with SVG effects.
   // Only non-null when ShouldBuildScrollInfoItemsForHoisting() is true.
